libcloudforensics/providers/azure/internal/monitoring.py

```diff
@@ -0,0 +1,135 @@
+                         'i.e. /subscriptions/<>/resourceGroups/<>/providers'
+                         '/<>/<>/yourResourceName'.format(resource_id))
+
+  def GetMetricsForResource(
+      self,
+      resource_id: str,
+      metrics: str,
+      from_date: Optional['datetime'] = None,
+      to_date: Optional['datetime'] = None,
+      interval: Optional[str] = None,
+      aggregation: str = 'Total',
+      qfilter: Optional[str] = None) -> Dict[str, List[str]]:
+    """Retrieve metrics for a given resource.
+
+    Args:
+      resource_id (str): The resource ID for which to lookup the metric.
+      metrics (str): A comma separated list of metrics to retrieve. E.g.
+          'Percentage CPU,Network In'.
+      from_date (datetime.datetime): Optional. A start date from which to get
+          the metric. If passed, toDate is also required.
+      to_date (datetime.datetime): Optional. An end date until which to get the
+          metric. If passed, fromDate is also required.
+      interval (str): An interval for the metrics, e.g. 'PT1H' will output
+          metric's values with one hour granularity.
+      aggregation (str): Optional. The type of aggregation for the metric's
+          values. Default is 'Total'. Possible values: 'Total', 'Average'.
+          Both can be retrieved if passed as a single string, separated by a
+          comma.
+      qfilter (str): Optional. A filter for the query. See
+          https://docs.microsoft.com/en-us/rest/api/monitor/metrics/list for
+          details about filtering.
+
+    Returns:
+      Dict[str, List[str]]]: A dictionary mapping the metric to a list of the
+          metric's values.
+
+    Raises:
+        RuntimeError: If the resource could not be found.
+    """
+    kwargs = {'metricnames': metrics, 'aggregation': aggregation}
+    if from_date and to_date:
+      timespan = '{0:s}/{1:s}'.format(from_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                                      to_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+      kwargs['timespan'] = timespan
+    if interval:
+      kwargs['interval'] = interval
+    try:
+      metrics_data = self.monitoring_client.metrics.list(
+          resource_id, filter=qfilter, **kwargs)
+    except models.ErrorResponseException:
+      raise RuntimeError(
+          'Could not fetch metrics {0:s} for resource {1:s}.  Please make '
+          'sure you specified the full resource ID  url, i.e. /subscriptions/'
+          '<>/resourceGroups/<>/providers/<>/<>/yourResourceName'.format(
+              metrics, resource_id))
+    results = {}  # type: Dict[str, List[str]]
+    for metric in metrics_data.value:
+      values = []
+      for timeserie in metric.timeseries:
+        for data in timeserie.data:
+          if data.time_stamp and data.total:
+            values.append('{0:s}: {1:s}'.format(str(data.time_stamp),
+                                                str(data.total)))
```

Comment:
Wouldn't it be better to output a dict here? We should probably leave formatting up to to whoever requests the data

Reply:
Updated